pkg/decode/decode.go

```diff
@@ -763,6 +763,26 @@ func (d *D) FieldGet(name string) *Value {
 	return nil
 }
 
+func (d *D) FieldGetScalar(name string) *scalar.S {
+	v := d.FieldGet(name)
+	if v == nil {
+		return nil
+	}
+	sr, ok := v.V.(*scalar.S)
+	if !ok {
+		panic("not a scalar value")
+	}
+	return sr
+}
+
+func (d *D) FieldTryGetScalarActualU(name string) (bool, uint64) {
```

Comment:
Not used?

I've tried to avoid accessing the decode tree in decoders if possible, it has usually been a sign of doing something fishy.  Also i have a feeling it might make it harder to change the decode internals in the future.